modules/javafx.graphics/src/test/java/test/javafx/css/StyleableProperty_transition_Test.java

```diff
@@ -119,63 +151,51 @@ public class StyleableProperty_transition_Test {
-        }
-
-        throw new AssertionError();
-    }
+    static StyleableStringProperty stringProperty;
+    static StyleableObjectProperty<Color> interpolatableObjectProperty;
+    static StyleableObjectProperty<Background> componentTransitionableObjectProperty;
 
     @SuppressWarnings("rawtypes")
-    record TestRun(StyleableProperty property, Object defaultValue, Object newValue) {}
+    record TestRun(StyleableProperty property, String fieldName, Object defaultValue, Object newValue) {}
 
     static Stream<TestRun> transitionParameters() {
         booleanProperty = new SimpleStyleableBooleanProperty(booleanPropertyMetadata, testBean, null);
         doubleProperty = new SimpleStyleableDoubleProperty(doublePropertyMetadata, testBean, null);
         floatProperty = new SimpleStyleableFloatProperty(floatPropertyMetadata, testBean, null);
         integerProperty = new SimpleStyleableIntegerProperty(integerPropertyMetadata, testBean, null);
         longProperty = new SimpleStyleableLongProperty(longPropertyMetadata, testBean, null);
-        objectProperty = new SimpleStyleableObjectProperty<>(objectPropertyMetadata, testBean, null, Color.RED);
+        stringProperty = new SimpleStyleableStringProperty(stringPropertyMetadata, testBean, null);
+        interpolatableObjectProperty = new SimpleStyleableObjectProperty<>(
+            interpolatableObjectPropertyMetadata, testBean, null, Color.RED);
+        componentTransitionableObjectProperty = new SimpleStyleableObjectProperty<>(
+            componentTransitionableObjectPropertyMetadata, testBean, null, Background.fill(Color.RED));
 
         return Stream.of(
-            new TestRun(booleanProperty, false, true),
-            new TestRun(doubleProperty, 0, 1),
-            new TestRun(floatProperty, 0, 1),
-            new TestRun(integerProperty, 0, 1),
-            new TestRun(longProperty, 0, 1),
-            new TestRun(objectProperty, Color.RED, Color.GREEN)
+            new TestRun(booleanProperty, "mediator", false, true),
+            new TestRun(doubleProperty, "mediator", 0, 1),
+            new TestRun(floatProperty, "mediator", 0, 1),
+            new TestRun(integerProperty, "mediator", 0, 1),
+            new TestRun(longProperty, "mediator", 0, 1),
+            new TestRun(stringProperty, "mediator", "foo", "bar"),
+            new TestRun(interpolatableObjectProperty, "controller", Color.RED, Color.GREEN),
+            new TestRun(componentTransitionableObjectProperty, "controller",
+                        Background.fill(Color.RED), Background.fill(Color.GREEN))
         );
     }
 
+    StubToolkit toolkit;
     Scene scene;
     Stage stage;
 
+    void setAnimationTime(long time) {
+        toolkit.setCurrentTime(time);
+        toolkit.handleAnimation();
+    }
+
     @BeforeEach
     void setup() {
-        scene = new Scene(new Group(testBean));
+        toolkit = (StubToolkit)Toolkit.getToolkit();
+        scene = new Scene(new Group());
```

Comment:
should we `hide` stage in `@AfterEach` ?
probably applies to other tests

Reply:
I'm already doing this in `@AfterEach` by calling `stage.close()`, which is equivalent to `hide()`.

Reply:
I should have looked in eclipse.  You are right - it is there.